### PR TITLE
Introduced a method, SwitchBaseURL(), enabling the selection of alter…

### PR DESCRIPTION
**Pull request by allanpk716**
Introduced a method, SwitchBaseURL(), enabling the selection of alternative API base URLs to accommodate varying network conditions across countries.

# Description

In my country, accessing api.themoviedb.org is restricted in many regions, while api.tmdb.org remains accessible. The proposed enhancement allows for switching the base API URL, such as with SwitchBaseURL(TMDB), prior to invoking Init().

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

All test cases from the original project have passed. Utilizing the new SwitchBaseURL(TMDB) function yields accurate results when querying data as well.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: authentication.go [hunk 1]
```diff
@@ -98,7 +98,7 @@ type SessionWithLogin struct {
 //
 // https://developers.themoviedb.org/3/authentication/validate-request-token
 // func (c *Client) CreateSessionWithLogin(u, p, rt string) (*RequestToken, error) {
-// 	tmdbURL := fmt.Sprintf("%s%stoken/validate_with_login?api_key=%s", baseURL, authenticationURL, c.apiKey)
+// 	tmdbURL := fmt.Sprintf("%s%stoken/validate_with_login?api_key=%s", hhddjgetBaseURL(), authenticationURL, c.apiKey)
```
Comment (cyruzin) [inline authentication.go:101]: hhddjgetBaseURL() ?

File: authentication.go [hunk 2]
```diff
@@ -120,7 +120,7 @@ type SessionWithLogin struct {
 //
 // https://developers.themoviedb.org/3/authentication/create-session-from-v4-access-token
 // func (c *Client) CreateSessionFromV4(at string) (*Session, error) {
-// 	tmdbURL := fmt.Sprintf("%s%ssession/convert/4?api_key=%s", baseURL, authenticationURL, c.apiKey)
+// 	tmdbURL := fmt.Sprintf("%s%ssession/convert/4?api_key=%s", hhddjgetBaseURL(), authenticationURL, c.apiKey)
```
Comment (cyruzin) [inline authentication.go:123]: here too: hhddjgetBaseURL() ?

**Review by cyruzin — CHANGES_REQUESTED** [review]
Can you implement like this?

```
// TMDb constants
const (
	defaultBaseURL    = "https://api.themoviedb.org/3"
	alternateBaseURL  = "https://api.tmdb.org/3"
	permissionURL     = "https://www.themoviedb.org/authenticate/"
	authenticationURL = "/authentication/"
	movieURL          = "/movie/"
	tvURL             = "/tv/"
	tvSeasonURL       = "/season/"
	tvEpisodeURL      = "/episode/"
	personURL         = "/person/"
	searchURL         = "/search/"
	collectionURL     = "/collection/"
	companyURL        = "/company/"
	configurationURL  = "/configuration/"
	creditURL         = "/credit/"
	discoverURL       = "/discover/"
	networkURL        = "/network/"
	keywordURL        = "/keyword/"
	genreURL          = "/genre/"
	guestSessionURL   = "/guest_session/"
	listURL           = "/list/"
	accountURL        = "/account/"
	watchProvidersURL = "/watch/providers/"
)

var baseURL = defaultBaseURL

// SetAlternateBaseURL sets an alternate base url.
func (c *Client) SetAlternateBaseURL() {
	baseURL = alternateBaseURL
}
```

**Comment by allanpk716** [comment]
Your improved solution is better, I think I can withdraw this is feature merge request. I'll resubmit it after I've reworked it.

**Pull request closed without merging**